synfig-studio/src/gui/canvasview.cpp

```diff
@@ -1453,114 +1454,205 @@ CanvasView::init_menus()
+	// );
 
+	Glib::RefPtr<Gio::SimpleAction> close_document_ = Gio::SimpleAction::create("close-document");
+	action_group->add_action("close-document",sigc::hide_return(sigc::mem_fun(*this,&CanvasView::close_instance)));
+
+	// action_group->add( Gtk::Action::create("quit", Gtk::StockID("gtk-quit"), _("Quit")),
+	// 	sigc::hide_return(sigc::ptr_fun(&App::quit))
+	// );
+
+	Glib::RefPtr<Gio::SimpleAction> quit_ = Gio::SimpleAction::create("quit");
+	action_group->add_action("quit",sigc::hide_return(sigc::ptr_fun(&App::quit)));
+
+	// action_group->add( Gtk::Action::create("select-all-ducks", _("Select All Handles")),
+	// 	sigc::mem_fun(*work_area,&WorkArea::select_all_ducks)
+	// );
+
+	Glib::RefPtr<Gio::SimpleAction> select_all_ducks_ = Gio::SimpleAction::create("select-all-ducks");
+	action_group->add_action("select-all-ducks",sigc::mem_fun(*work_area,&WorkArea::select_all_ducks));
+
+	// action_group->add( Gtk::Action::create("unselect-all-ducks", _("Unselect All Handles")),
+	// 	sigc::mem_fun(*work_area,&WorkArea::unselect_all_ducks)
+	// );
+
+	Glib::RefPtr<Gio::SimpleAction> unselect_all_ducks_ = Gio::SimpleAction::create("unselect-all-ducks");
+	action_group->add_action("unselect-all-ducks",sigc::mem_fun(*work_area,&WorkArea::unselect_all_ducks));
+
+	// action_group->add( Gtk::Action::create("select-all-layers", _("Select All Layers")),
+	// 	sigc::mem_fun(*this,&CanvasView::on_select_layers)
+	// );
+
+	Glib::RefPtr<Gio::SimpleAction> select_all_layers_ = Gio::SimpleAction::create("select-all-layers");
+	action_group->add_action("select-all-layers",sigc::mem_fun(*this,&CanvasView::on_select_layers));
+
+	// action_group->add( Gtk::Action::create("unselect-all-layers", _("Unselect All Layers")),
+	// 	sigc::mem_fun(*this,&CanvasView::on_unselect_layers)
+	// );
+
+	Glib::RefPtr<Gio::SimpleAction> unselect_all_layers_ = Gio::SimpleAction::create("unselect-all-layers");
+	action_group->add_action("unselect-all-layers",sigc::mem_fun(*this,&CanvasView::on_unselect_layers));
+
+	// action_group->add( Gtk::Action::create("pause", Gtk::StockID("synfig-animate_pause")),
+	// 	sigc::mem_fun(*this, &CanvasView::stop_async)
+	// );
+
+	Glib::RefPtr<Gio::SimpleAction> pause_ = Gio::SimpleAction::create("pause");
+	action_group->add_action("pause",sigc::mem_fun(*this, &CanvasView::stop_async));
+
+	// action_group->add( Gtk::Action::create("refresh", Gtk::StockID("gtk-refresh")),
+	// 	SLOT_EVENT(EVENT_REFRESH)
+	// );
+
+	Glib::RefPtr<Gio::SimpleAction> refresh_ = Gio::SimpleAction::create("refresh");
+	action_group->add_action("refresh",SLOT_EVENT(EVENT_REFRESH));
+
+	// action_group->add( Gtk::Action::create("properties", Gtk::StockID("gtk-properties"), _("Properties...")),
+	// 	sigc::mem_fun0(canvas_properties,&CanvasProperties::present)
+	// );
+
+	Glib::RefPtr<Gio::SimpleAction> properties = Gio::SimpleAction::create("properties");
+	action_group->add_action("properties",sigc::mem_fun0(canvas_properties,&CanvasProperties::present));
```

Comment:
Hm… the `properties` variable is useless here. You just create it. Same for all previous cases.

Are you testing the code?

Reply:
currently build is failing ! So, I won't be able to test but I am keep building it again and again until the error goes away ! 🙂 btw I am stuck on radio actions from yesterday ! They are implemented totally differently ! So, trying to understand that from yesterday night ! Moving forward for the time being !

Reply:
np ! I thought its a good practice to save that action in a different variable ! but NP i will remove those lines 🙂

Reply:
That isn't how you solve it.
You are creating an object and not using it. And, according to your code, you don't have to create that instance at all.

https://developer-old.gnome.org/gtkmm-tutorial/3.18/sec-actions.html

Reply:
> currently build is failing ! So, I won't be able to test but I am keep building it again and again until the error goes away !

The building error can go away, but logical errors may arise. You are changing too much without testing. :(

Reply:
> The building error can go away, but logical errors may arise. You are changing too much without testing. :(

I was afraid of this ! but everything is interconnected and I have to switch those parts that are interconnected ! Kindly guide me a bit, how should I go forward ? Thanks :) 

I will remove changes I've done till now ! No problem but kindly give me a right direction and a green signal to move forward 😄

Thanks :)

Reply:
> 
> 
> That isn't how you solve it.
> You are creating an object and not using it. And, according to your code, you don't have to create that instance at all.
> 
> https://developer-old.gnome.org/gtkmm-tutorial/3.18/sec-actions.html

Sorry, My bad ! I will check these out again :) No issues.

Reply:
@rodolforg I still don't understand !! How is it working ? without creating a new instance and by just referring to a particular name. Is it because of the fact that those actions already exists somewhere ??

Reply:
` 	Glib::RefPtr<Gio::SimpleAction> properties = Gio::SimpleAction::create("properties"); `
` 	Gio::SimpleAction::create("properties"); `
Both statements above do nothing. You create the actions but they are not linked/used anywhere.

` 	action_group->add_action("properties",sigc::mem_fun0(canvas_properties,&CanvasProperties::present)); `
If you read the documentation, you'll see the above method _creates_ a `SimpleAction`, bind it to the provided callback and add it to action group.
> A convenience method for creating a SimpleAction instance and adding it to the ActionMap.

Reply:
Understood ! Thanks.

Reply:
Understood !